app/src/main/java/io/homeassistant/companion/android/settings/notification/NotificationHistoryFragment.kt

```diff
@@ -17,69 +22,99 @@ import java.util.GregorianCalendar
+        super.onPrepareOptionsMenu(menu)
+        menu.setGroupVisible(R.id.notification_toolbar_group, true)
+
+        menu.findItem(R.id.get_help)?.let {
+            it.isVisible = true
+            it.intent = Intent(Intent.ACTION_VIEW, Uri.parse("https://companion.home-assistant.io/docs/notifications/notifications-basic"))
+        }
+        menu.findItem(R.id.last25)?.title = getString(R.string.last_num_notifications, 25)
+        menu.findItem(R.id.last50)?.title = getString(R.string.last_num_notifications, 50)
+        menu.findItem(R.id.last100)?.title = getString(R.string.last_num_notifications, 100)
+
+        val prefCategory = findPreference<PreferenceCategory>("list_notifications")
+        val notificationDao = AppDatabase.getInstance(requireContext()).notificationDao()
+        val allNotifications = notificationDao.getAll()
+
+        if (allNotifications.isNullOrEmpty()) {
+            menu.removeItem(R.id.search_notifications)
+            menu.removeItem(R.id.notification_filter)
+            menu.removeItem(R.id.action_delete)
+        } else {
+            val searchViewItem = menu.findItem(R.id.search_notifications)
+            val searchView: SearchView = MenuItemCompat.getActionView(searchViewItem) as SearchView
+            searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+                override fun onQueryTextSubmit(query: String?): Boolean {
+                    searchView.clearFocus()
+
+                    return false
+                }
+
+                override fun onQueryTextChange(query: String?): Boolean {
+                    var searchList: Array<NotificationItem> = emptyArray()
+                    if (!query.isNullOrEmpty()) {
+                        for (item in allNotifications) {
+                            if (item.message.contains(query, true))
+                                searchList += item
+                        }
+                        prefCategory?.title = getString(R.string.search_results)
+                        reloadNotifications(searchList, prefCategory)
+                    } else if (query.isNullOrEmpty()) {
+                        prefCategory?.title = getString(R.string.notifications)
+                        filterNotifications(filterValue, notificationDao, prefCategory)
+                    }
+                    return false
+                }
+            })
+        }
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        val prefCategory = findPreference<PreferenceCategory>("list_notifications")
+        val notificationDao = AppDatabase.getInstance(requireContext()).notificationDao()
+        if (item.itemId in listOf(R.id.last25, R.id.last50, R.id.last100)) {
+            filterValue = when (item.itemId) {
+                R.id.last25 -> "last25"
+                R.id.last50 -> "last50"
+                R.id.last100 -> "last100"
+                else -> "last25"
+            }
```

Comment:
Rather than return a string here, then in `filterNotifications` turn it into a number why not just pass the number of messages to filter?

Reply:
as usual a much cleaner method 😃  thank you!